### PR TITLE
Add current executed preProcess requests metric

### PR DESCRIPTION
**Pull request by yontyon**
Add metrics (gauge) that counts how many pre-execution requests are currently executed.

### CODE REVIEW

File: bftengine/src/preprocessor/PreProcessor.cpp [hunk 1]
```diff
@@ -113,7 +113,8 @@ PreProcessor::PreProcessor(shared_ptr<MsgsCommunicator> &msgsCommunicator,
                            metricsComponent_.RegisterCounter("preProcConsensusNotReached"),
                            metricsComponent_.RegisterCounter("preProcessRequestTimedout"),
                            metricsComponent_.RegisterCounter("preProcReqSentForFurtherProcessing"),
-                           metricsComponent_.RegisterCounter("preProcPossiblePrimaryFaultDetected")},
+                           metricsComponent_.RegisterCounter("preProcPossiblePrimaryFaultDetected"),
+                           metricsComponent_.RegisterGauge("currentPreProcReqNum", 0)},
```
Comment (yuliasherman) [inline bftengine/src/preprocessor/PreProcessor.cpp:117]: All other metrics have "preProc" prefix. Please rename it to preProcInFlyRequestsNum.

File: bftengine/src/preprocessor/PreProcessor.hpp [hunk 2]
```diff
@@ -155,6 +155,7 @@ class PreProcessor {
     concordMetrics::CounterHandle preProcessRequestTimedout;
     concordMetrics::CounterHandle preProcReqSentForFurtherProcessing;
     concordMetrics::CounterHandle preProcPossiblePrimaryFaultDetected;
+    concordMetrics::GaugeHandle currentPreProcReqNum;
```
Comment (yuliasherman) [inline bftengine/src/preprocessor/PreProcessor.hpp:158]: Please rename as mentioned before

File: util/include/Metrics.hpp [hunk 3]
```diff
@@ -68,6 +68,8 @@ class Gauge {
  public:
   explicit Gauge(const uint64_t val) : val_(val) {}
 
+  void Inc() { val_++; }
```
Comment (yuliasherman) [inline util/include/Metrics.hpp:71]: Nit: ++val_

File: util/include/Metrics.hpp [hunk 4]
```diff
@@ -68,6 +68,8 @@ class Gauge {
...
+  void Dec() { val_--; }
```
Comment (yuliasherman) [inline util/include/Metrics.hpp:72]: Nit: --val_

File: bftengine/src/preprocessor/PreProcessor.cpp [hunk 5]
```diff
@@ -368,6 +369,7 @@ void PreProcessor::cancelPreProcessing(NodeIdType clientId) {
       reqSeqNum = clientEntry->reqProcessingStatePtr->getReqSeqNum();
       releaseClientPreProcessRequest(clientEntry, clientId, CANCEL);
       LOG_WARN(logger(), "Pre-processing consensus not reached - abort request " << KVLOG(reqSeqNum, clientId));
+      preProcessorMetrics_.currentPreProcReqNum.Get().Dec();
```
Comment (yuliasherman) [inline bftengine/src/preprocessor/PreProcessor.cpp:372]: Please increment this metric in registerRequest and decrement in releaseClientPreProcessRequest. No other places should be changed :-)

File: bftengine/src/preprocessor/PreProcessor.cpp (hunk 1 above)
Comment (yontyon) [reply bftengine/src/preprocessor/PreProcessor.cpp]: done

File: bftengine/src/preprocessor/PreProcessor.hpp (hunk 2 above)
Comment (yontyon) [reply bftengine/src/preprocessor/PreProcessor.hpp]: done

File: util/include/Metrics.hpp (hunk 3 above)
Comment (yontyon) [reply util/include/Metrics.hpp]: done

File: util/include/Metrics.hpp (hunk 4 above)
Comment (yontyon) [reply util/include/Metrics.hpp]: done

File: bftengine/src/preprocessor/PreProcessor.cpp [hunk 6]
```diff
@@ -114,7 +114,7 @@ PreProcessor::PreProcessor(shared_ptr<MsgsCommunicator> &msgsCommunicator,
                            metricsComponent_.RegisterCounter("preProcessRequestTimedout"),
                            metricsComponent_.RegisterCounter("preProcReqSentForFurtherProcessing"),
                            metricsComponent_.RegisterCounter("preProcPossiblePrimaryFaultDetected"),
-                           metricsComponent_.RegisterGauge("currentPreProcReqNum", 0)},
+                           metricsComponent_.RegisterGauge("PreProcCurrentReqNum", 0)},
```
Comment (yuliasherman) [inline bftengine/src/preprocessor/PreProcessor.cpp:117]: preProcCurrentReqNum sounds like a "sequence number of current request", which is confusing.

PreProcCurrentReqNum -> preProcInFlyRequestsNum

Comment (yontyon) [reply bftengine/src/preprocessor/PreProcessor.cpp]: fixed

**Pull request merged**